KeyError: 'ETag'

**What happened**:
I get a KeyError when trying to copy files from bucket to local:
```
>       _, _, parts_suffix = info["ETag"].strip('"').partition("-")
E       KeyError: 'ETag'
```

**What you expected to happen**:
files get copied

**Minimal Complete Verifiable Example**:

```python
from s3fs import S3FileSystem
s3 = S3FileSystem()
s3.copy(input_path, '/tmp/foobar', recursive=True)
```

**Anything else we need to know?**:
in the _info() function, when the ls was cached, the response never includes the 'Etag' key, so I think it's a bug to rely on it always being there.

**Environment**:

- s3fs version: 2021.4.0
- Python version: 3.6.10
- Operating System: Mac
- Install method (conda, pip, source): pip

--- a/s3fs/core.py
+++ b/s3fs/core.py
@@ -1518,7 +1518,7 @@ class S3FileSystem(AsyncFileSystem):
         info = await self._info(path1, bucket, key, version_id=vers)
         size = info["size"]
 
-        _, _, parts_suffix = info["ETag"].strip('"').partition("-")
+        _, _, parts_suffix = info.get("ETag", "").strip('"').partition("-")
         if preserve_etag and parts_suffix:
             await self._copy_etag_preserved(
                 path1, path2, size, total_parts=int(parts_suffix)

--- a/s3fs/tests/test_s3fs.py
+++ b/s3fs/tests/test_s3fs.py
@@ -2079,3 +2079,19 @@ def test_lsdir(s3):
 
     d = test_bucket_name + "/test"
     assert d in s3.ls(test_bucket_name)
+
+
+def test_copy_file_without_etag(s3, monkeypatch):
+
+    s3.touch(test_bucket_name + "/copy_tests/file")
+    s3.ls(test_bucket_name + "/copy_tests/")
+
+    [file] = s3.dircache[test_bucket_name + "/copy_tests"]
+
+    assert file["name"] == test_bucket_name + "/copy_tests/file"
+    file.pop("ETag")
+
+    assert s3.info(file["name"]).get("ETag", None) is None
+
+    s3.cp_file(file["name"], test_bucket_name + "/copy_tests/file2")
+    assert s3.info(test_bucket_name + "/copy_tests/file2")["ETag"] is not None

